---
title: C6310 | Microsoft Docs
ms.custom: ''
ms.date: 11/15/2016
ms.prod: visual-studio-dev14
ms.reviewer: ''
ms.suite: ''
ms.technology:
- vs-devops-test
ms.tgt_pltfrm: ''
ms.topic: article
f1_keywords:
- C6310
helpviewer_keywords:
- C6310
ms.assetid: e759eb63-883f-4c3e-bf2f-b924ff511405
caps.latest.revision: 15
author: mikeblome
ms.author: mblome
manager: ghogen
ms.openlocfilehash: 8bbe62582490f45c5bc1a5a7eb6a2c59820a4074
ms.sourcegitcommit: af428c7ccd007e668ec0dd8697c88fc5d8bca1e2
ms.translationtype: MT
ms.contentlocale: fr-FR
ms.lasthandoff: 11/16/2018
ms.locfileid: "51800742"
---
# <a name="c6310"></a>C6310
[!INCLUDE[vs2017banner](../includes/vs2017banner.md)]

avertissement C6310 : une constante non autorisée dans le filtre d’exception peut provoquer un comportement inattendu  
  
 Ce message indique qu’une constante non autorisée a été détectée dans l’expression de filtre d’un gestionnaire d’exceptions structurées. Les constantes définies pour une utilisation dans l’expression de filtre d’un gestionnaire d’exceptions structurées sont :  
  
- `EXCEPTION_CONTINUE_EXECUTION`  
  
- `EXCEPTION_CONTINUE_SEARCH`  
  
- `EXCEPTION_EXECUTE_HANDLER`  
  
  Ces valeurs sont définies dans le excpt.h de fichier d’en-tête runtime.  
  
  À l’aide d’une constante qui n’est pas dans la liste précédente peut provoquer un comportement inattendu.  
  
## <a name="example"></a>Exemple  
 Le code suivant génère cet avertissement :  
  
```  
#include <excpt.h>  
#include <stdio.h>  
#include <windows.h>  
  
BOOL LimitExceeded();  
  
void fd( )  
{  
  __try   
  {  
    if (LimitExceeded())   
    {  
      RaiseException(EXCEPTION_ACCESS_VIOLATION,0,0,0);  
    }  
    else  
    {  
      // code   
    }  
  }   
  __except (EXCEPTION_ACCESS_VIOLATION)  
  {  
    puts("Exception Occurred");  
  }  
}  
```  
  
 Pour corriger cet avertissement, utilisez le code suivant :  
  
```  
#include <excpt.h>  
#include <stdio.h>  
#include <windows.h>  
  
BOOL LimitExceeded();  
  
void fd( )  
{  
  __try   
  {  
    if (LimitExceeded())   
    {  
      RaiseException(EXCEPTION_ACCESS_VIOLATION,0,0,0);  
    }  
    else  
    {  
      // code   
    }  
  }   
  __except (GetExceptionCode()==EXCEPTION_ACCESS_VIOLATION ?   
               EXCEPTION_EXECUTE_HANDLER : EXCEPTION_CONTINUE_SEARCH)  
  {  
    puts("Exception Occurred");  
  }  
}  
```



